src/infra/dex/okx/dto.rs

```diff
@@ -180,10 +225,46 @@ pub struct ApproveTransactionRequest {
 
 impl ApproveTransactionRequest {
     pub fn with_domain(chain_index: u64, order: &dex::Order) -> Self {
+        // For buy orders (ExactOut mode), we need to use U256::MAX allowance
+        // because the exact input amount is not known in advance.
+        let approve_amount = if matches!(order.side, order::Side::Buy) {
+            U256::MAX
+        } else {
+            order.amount.get()
+        };
```

Comment:
Could you measure how much time we'd lose if we don't do the approve and quote request in parallel? Given that this is not doing any complicated computation it should be a few ms at most. In that case I think it would be nicer to restructure the code to build the approve request with the amounts suggested by the quote.